### PR TITLE
feat: New arch Android

### PR DESCRIPTION
**Pull request by Titozzz**

### CODE REVIEW

File: src/WebViewTypes.ts [hunk 1]
```diff
@@ -251,7 +247,7 @@ export interface WebViewNativeConfig {
   component?:
     | typeof NativeWebViewIOS
     | typeof NativeWebViewMacOS
-    | typeof NativeWebViewAndroid;
+    | typeof NativeWebViewComponent;
```
Comment (AndreiCalazans) [inline src/WebViewTypes.ts:250]: Won't Fabric force normalization to a single interface? Are you keeping iOS and MacOS until it's fully migrated?

**Comment by Titozzz** [comment]
> Excuse me, is there any new progress on this. We urgently need it now

If you are in a hurry, either:
- ask what can do done to help on the code
- sponsor people working on that code
- don't use fabric right now cause it's still very early

**Comment by TheAlmightyBob** [comment]
@Titozzz Are there particular changes to pay attention to here? For example, it's hard to tell if the native code changed other than being extracted into separate files for the different classes (which certainly makes sense and I wonder if that should just be its own PR...)

Do you expect this to be a breaking change (e.g. require a new version of RN or enabling the new architecture flags in apps that want to use the latest version of `react-native-webview`)?

**Comment by Titozzz** [comment]
@TheAlmightyBob it won't be breaking. I'm trying to get the new arch to work the same way as the old arch, but old arch users shouldn't be afraid 😅

**Comment by alfonsocj** [comment]
@Titozzz do you have an estimate on when you will land this? I want to give supporting fabric on iOS a try but I want to leverage your codegen changes 😄

**Comment by Titozzz** [comment]
I've already done iOS 😅, you can find my branch new-arch-ios. It only needs polishing with view lifecycle

**Comment by alfonsocj** [comment]
Nice! Is there anything I can do to help you polish it?

**Comment by kartikmittal1996** [comment]
> _ No description provided. _

Can I use this branch in my project with new architecture and fabric enabled?

**Comment by Titozzz** [comment]
I wouldn't use it in production yet as it's not released, but you can try, also I'd start from the iOS branch that contains android and has fixes

**Comment by Titozzz** [comment]
Closed in favor of https://github.com/react-native-webview/react-native-webview/pull/2686 containing both platforms

**Pull request closed without merging**